Full Migration to Python3.x complete

asrabon opened the pull request:
The code base has been full migrated to Python 3.x. The functionality of the code is all the same however, some modifications have been made to simplify or refactor sections.

asrabon committed:
Full Migration to Python3.x complete

TryCatchHCF commented:
Thanks Asrabon, I'll review and work on the merge over the holidays. Hope
you and yours are well.

-TryCatchHCF

On Mon, Nov 16, 2020 at 9:53 AM asrabon <notifications@github.com> wrote:

> The code base has been full migrated to Python 3.x. The functionality of
> the code is all the same however, some modifications have been made to
> simplify or refactor sections.
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/TryCatchHCF/Cloakify/pull/9
> Commit Summary
>
>    - Full Migration to Python3.x complete
>
> File Changes
>
>    - *A* .gitignore
>    <https://github.com/TryCatchHCF/Cloakify/pull/9/files#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947>
>    (1)
>    - *M* cloakify.py
>    <https://github.com/TryCatchHCF/Cloakify/pull/9/files#diff-364dc3e112b299b334688ed5182faba7126ef82631d80cc9615197e6be0651ba>
>    (62)
>    - *M* cloakifyFactory.py
>    <https://github.com/TryCatchHCF/Cloakify/pull/9/files#diff-060c3eece52c023a6021635b6c8c90244fad2815bb1293103029adadb4b6345a>
>    (700)
>    - *M* decloakify.py
>    <https://github.com/TryCatchHCF/Cloakify/pull/9/files#diff-c64fdbfcb0144baa2ff7acee255d4a59fd175207f7c3a744c8421c5b7a2683ae>
>    (45)
>    - *A* noiseTools/__init__.py
>    <https://github.com/TryCatchHCF/Cloakify/pull/9/files#diff-14056af97beb44483a9ac3415a40136eda26aef523195a0388d0fec4231757a8>
>    (4)
>    - *M* noiseTools/prependEmoji.py
>    <https://github.com/TryCatchHCF/Cloakify/pull/9/files#diff-3199798ba01a0e9c1c53e5fcc137695ec8033006e2c43c39f48cfe26711fb9d6>
>    (46)
>    - *M* noiseTools/prependID.py
>    <https://github.com/TryCatchHCF/Cloakify/pull/9/files#diff-ed7c0314f81849450394afcebdf2975630637e01dc2e8b0892e9b8787c4ae365>
>    (49)
>    - *M* noiseTools/prependLatLonCoords.py
>    <https://github.com/TryCatchHCF/Cloakify/pull/9/files#diff-f75c7c6a55829e698c90c390bae26a39bdc0ff01d84db2c2e9926ffb5d42ff94>
>    (65)
>    - *M* noiseTools/prependTimestamps.py
>    <https://github.com/TryCatchHCF/Cloakify/pull/9/files#diff-89b94ec056961d0efdc070a8a884b19c5b35196a04b8718c2421d27cf475167a>
>    (70)
>    - *M* removeNoise.py
>    <https://github.com/TryCatchHCF/Cloakify/pull/9/files#diff-01093adccc5a0ac029d38e97d983088f005e2260bae1955c6210ba063e89202c>
>    (24)
>
> Patch Links:
>
>    - https://github.com/TryCatchHCF/Cloakify/pull/9.patch
>    - https://github.com/TryCatchHCF/Cloakify/pull/9.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/TryCatchHCF/Cloakify/pull/9>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEK363K2DOID7NVWNE6JVBDSQFKI7ANCNFSM4TXNJ5ZQ>
> .
>

asrabon commented:
Awesome I have been a huge fan of this repository for a while and wanted to help out and contribute! 

I do a lot of research into steganography and had other ideas of features that could be integrated to increase the level of obfuscation but, wanted to get the baseline stuff done first like migrating it to Python 3.x.

asrabon committed:
Create codeql-analysis.yml

asrabon committed:
Revert "Create codeql-analysis.yml"

This reverts commit 645f1ea8fb605620086c9f92b3cec7127b4434e5.

asrabon committed:
Added ability to password protect cloakified file

Added the ability in cloakifyFactory.py that prompts the user to use a password to protect the cloaked file. This will simply set the seed of the random library to the password the user provided, then shuffle the order the lines are written in. For decoding it sets the seed of the random library to the password the user provides and preforms the same shuffling process so it knows how to reconstruct the original base64 payload

asrabon committed:
Create lint-and-test.yml

Create github workflow to install any dependencies, lint with flake8, and test on python 3.5+

asrabon committed:
Update lint-and-test.yml

Removed Python 3.5 from version testing since code contains fstrings

asrabon committed:
Created tests and fixed ciphers

Created tests to validate all ciphers gave the proper output when cloaked and decloaked, configured the same test for cipher cloaking with passwords, and created a test to check for duplicate lines in ciphers. Fixed 2 ciphers that contained duplicate lines which led to the decloaked file not matching the original payload due to multiple mappings

asrabon committed:
Merge branch 'master' of https://github.com/asrabon/Cloakify

asrabon committed:
Fixed Pytest Bug & Added to gitignore

Zagreus9723 commented:
@TryCatchHCF Can u merge? or @asrabon can u upload this to another repo?

p1r473 commented:
@TryCatchHCF Can you take a look at merging? Thanks!

asrabon commented:
@Apophis52 @p1r473 I have moved my code over to my own repository since @TryCatchHCF never reviewed and merged my PR.

Along with the port to Python 3.x I have also added the ability to password protect a file as well. If you have any other suggestions or features to add please put them in my repo.
https://github.com/asrabon/Cloakify-3

aniketkumar2510 commented:
> @Apophis52 @p1r473 I have moved my code over to my own repository since @TryCatchHCF never reviewed and merged my PR.
> 
> Along with the port to Python 3.x I have also added the ability to password protect a file as well. If you have any other suggestions or features to add please put them in my repo. https://github.com/asrabon/Cloakify-3

hey @asrabon I was trying out your cloakify for python 3 and turns out that I am not able to cloak my file as after cloaking it just shows me a blank file. I was wondering if you could help me fix it thanks!

p1r473 commented:
@TryCatchHCF bump

asrabon commented:
@aniketkumar2510 Could you describe the process you go through and what options you select? I am not able to reproduce on my side and running cloakifyFactory.py with various different inputs has all generated a valid output.
